charts/opentelemetry-collector/templates/service.yaml

```diff
@@ -12,12 +12,15 @@ metadata:
   {{- end }}
 spec:
   type: {{ .Values.service.type }}
-  {{- if .Values.service.clusterIP }}
+  {{- if and .Values.service.clusterIP (eq .Values.service.type "ClusterIP") }}
```

Comment:
I don't think we need these extra restrictions.  The [k8s load balancer service example](https://kubernetes.io/docs/concepts/services-networking/service/#loadbalancer) shows that LoadBalancer type and ClusterIP are not mutually exclusive.  For each of these fields lets stick with only checking if the field is set.